tests/testthat/test-finalsize_by_group.R

```diff
@@ -0,0 +1,292 @@
+test_that("Check final size calculation is correct in simple case", {
+  r0 <- 2
+  contact_matrix <- matrix(r0 / 200.0, 2, 2)
+  demography_vector <- rep(100.0, 2) |> as.matrix()
+  psusc <- rep(1.0, 2) |> as.matrix()
+  susc <- rep(1.0, 2) |> as.matrix()
+
+  contact_matrix = apply(
+    contact_matrix, 1, function(r) r / demography_vector
+  )
+
+  epi_outcome <- final_size_grps_cpp(
+    contact_matrix = contact_matrix,
+    demography_vector = demography_vector,
+    p_susceptibility = psusc,
+    susceptibility = susc
+  )
+  
+  testthat::expect_vector(
+    epi_outcome, ptype = numeric()
+  )
+  testthat::expect_false(
+    any(is.nan(epi_outcome))
+  )
+  testthat::expect_false(
+    any(is.infinite(epi_outcome))
+  )
+  testthat::expect_true(
+    all(epi_outcome > 0.0)
+  )
+
+  epi_outcome_known <- 1 - exp(-r0 * epi_outcome)
+
+  testthat::expect_equal(
+    epi_outcome, epi_outcome_known
+  )
+})
+
+test_that("Check final size by groups works with multiple risk groups", {
+  contact_matrix = matrix(c(1, 0.5, 0.5, 1), 2, 2)
+  demography_vector = rep(1, 2)
+  p_susceptibility = matrix(c(0.25, 0.75, 0.25, 0.75), 2, 2)
```

Comment:
Rows should sum up to 1 (might be good to add this as a check to the final_size_grps_cpp?)

```
p_susceptibility = matrix(c(0.25, 0.25, 0.75, 0.75), 2, 2)
```

Reply:
May be worth constructing test matrices via rbind going forward to avoid confusion. i.e.
``` r
dat <- matrix(c(0.25, 0.25, 0.75, 0.75), 2, 2)
dat2 <- rbind(
    c(0.25, 0.75),
    c(0.25, 0.75)
)
identical(dat, dat2)
#> [1] TRUE
```

Reply:
Fixed `p_susceptibility` rowwise sums in tests.

Added check for rowwise sum within `final_size_grps_cpp` in https://github.com/epiverse-trace/finalsize/pull/39/commits/5a16c3a387f88f064d23edd2c77c77f276324968